sources/metricdog/src/host_check.rs

```diff
@@ -0,0 +1,15 @@
+use crate::error::Result;
+
+pub(crate) trait HostCheck {
+    /// Checks whether this current boot is the first boot of the host.
+    fn is_first_boot(&self) -> Result<bool>;
+
+    /// The time, in milliseconds, it took for the host to reach the 'preconfigured' stage.
+    fn preconfigured_time(&self) -> Result<String>;
+
+    /// The time, in milliseconds, it took for the host to reach the 'configured' stage.
+    fn configured_time(&self) -> Result<String>;
```

Comment:
Do you need one as well for `multi-user.target`? The orchestrators' agents are started in this target, they could take a while to initialize, and we may want to know why.

Reply:
`send-boot-success.service` runs before `multi-user.target` so I would need to capture this event separately and have a separate subcommand for it. I believe in offline discussions we decided to not track orchestrator agent start times for now since there are too many factors involved with service start times in `multi-user.target`. Instead we'll track parts of the boot that are more consistent and not as heavily influenced by different configuration.

Reply:
We can also move `send-boot-success` to `multi-user.target` and have it run after `kubelet`/`ecs-agent` which means we would no longer send a boot success if the orchestration agent fails. I think that change makes sense but I would like to head some opinions on that.

Reply:
> Instead we'll track parts of the boot that are more consistent and not as heavily influenced by different configuration.

Ok, I agree with this, thanks for the clarification.